ytypes/util_schema.go

```diff
@@ -155,10 +155,16 @@ func directDescendantSchema(f reflect.StructField) (string, error) {
 	if err != nil {
 		return "", err
 	}
+
+	// Early return for performance.
+	if !strings.Contains(pathAnnotation, "|") && !strings.Contains(pathAnnotation, "/") {
+		return pathAnnotation, nil
+	}
+
 	paths := strings.Split(pathAnnotation, "|")
 
 	for _, pth := range paths {
-		if len(strings.Split(pth, "/")) == 1 {
+		if !strings.Contains(pth, "/") {
 			return pth, nil
 		}
 	}
```

Comment:
Does this actually improve performance? Split and Contains both do a full scan of the string right? I'm ok with keeping this, just it does make the code slightly harder to read.

Reply:
Here's the `genSplit` code. It also *must* pass through the entire string as oppose to `strings.Contains` can early stop at the first occurrence:

```go
// Generic split: splits after each instance of sep,
// including sepSave bytes of sep in the subarrays.
func genSplit(s, sep string, sepSave, n int) []string {
	if n == 0 {
		return nil
	}
	if sep == "" {
		return explode(s, n)
	}
	if n < 0 {
		n = Count(s, sep) + 1
	}

	if n > len(s)+1 {
		n = len(s) + 1
	}
	a := make([]string, n)
	n--
	i := 0
	for i < n {
		m := Index(s, sep)
		if m < 0 {
			break
		}
		a[i] = s[:m+sepSave]
		s = s[m+len(sep):]
		i++
	}
	a[i] = s
	return a[:i+1]
}
```

Reply:
I was considering the case where no separators "|" or "/" were present, since that is the only case where the early return does save time. In this case I think not much heap allocations need to be made for adding new path elements, but I agree it still saves that allocation.